src/data/fintype/card_embedding.lean

```diff
@@ -50,29 +48,29 @@ begin
     { rw [finset.card_compl, finset.card_map, finset.card_fin] },
     { simp } },
 
-  -- putting `card_sigma` in `simp` causes it to not fully simplify
-  rw card_sigma,
-  simp only [this, finset.sum_const, finset.card_univ, nsmul_eq_mul, nat.cast_id],
-
-  replace h := (nat.lt_of_succ_le h).le,
-  rw [nat.desc_factorial_succ, hn h, mul_comm]
+  simp only [this, card_sigma, finset.sum_const, finset.card_univ, nsmul_eq_mul, nat.cast_id],
+  rw [nat.desc_factorial_succ, hn (nat.lt_of_succ_le h).le, mul_comm]
 end
 
+local attribute [irreducible] function.embedding.fintype
```

Comment:
What is this needed for?

Reply:
I need it `semireducible` for the `nontriviality` to work, but the `convert` in `card_embedding_eq` works too hard trying to match up all the `fintype` instances if it's not again irreducible.

Reply:
Can you add a comment to that effect?